otter/test/convergence/test_service.py

```diff
@@ -1,95 +1,425 @@
-from effect import Constant, Effect, ParallelEffects, parallel
-from effect.testing import Stub
+from functools import partial
+
+from effect import (
+    ComposedDispatcher, Constant, Effect, Error, Func, ParallelEffects,
+    TypeDispatcher, base_dispatcher, sync_perform)
+from effect.async import perform_parallel_async
+from effect.ref import Reference, reference_dispatcher
+from effect.testing import EQDispatcher, EQFDispatcher
 
 import mock
 
 from pyrsistent import freeze, pmap, pset, s
 
-from twisted.internet.defer import fail
-from twisted.internet.task import Clock
+from twisted.internet.defer import fail, succeed
 from twisted.trial.unittest import SynchronousTestCase
 
-from otter.constants import ServiceType
+from otter.constants import CONVERGENCE_DIRTY_DIR, ServiceType
 from otter.convergence.model import (
-    CLBDescription, CLBNode, NovaServer, ServerState)
+    CLBDescription, CLBNode, NovaServer, ServerState, StepResult)
 from otter.convergence.service import (
-    Converger, determine_active, execute_convergence)
-from otter.http import TenantScope, service_request
-from otter.models.intents import ModifyGroupState
-from otter.models.interface import GroupState
+    ConcurrentError,
+    ConvergenceStarter,
+    Converger,
+    converge_all,
+    converge_one,
+    determine_active, execute_convergence, get_my_divergent_groups,
+    make_lock_set,
+    mark_divergent,
+    non_concurrently)
+from otter.http import service_request
+from otter.models.intents import (
+    GetScalingGroupInfo, ModifyGroupState, perform_modify_group_state)
+from otter.models.interface import GroupState, NoSuchScalingGroupError
 from otter.test.convergence.test_planning import server
+from otter.test.util.test_zk import ZNodeStatStub
 from otter.test.utils import (
-    CheckFailure, LockMixin, mock_group, mock_log, resolve_effect,
-    resolve_stubs)
-from otter.util.fp import assoc_obj
+    CheckFailureValue, FakePartitioner, IsBoundWith,
+    matches,
+    mock_group, mock_log,
+    transform_eq)
+from otter.util.zk import CreateOrSet, DeleteNode, GetChildrenWithStats
+
+
+class MarkDivergentTests(SynchronousTestCase):
```

Comment:
Totally non-blocking - I am generally all in favor of more test coverage, but this seems like a duplicate test of `test_start_convergence`, since `mark_divergent` seems more like an implementation detail of `start_convergence`?